src/main/java/net/mcreator/ui/action/impl/workspace/SDKManager.java

```diff
@@ -67,6 +71,13 @@ public static void downloadSDK(GeneratorConfiguration generatorConfiguration,
 				progressUnit.getProgressDialog().refreshDisplay();
 
 			getJDKFolderForVersion(jdkVersion + fileExtension).delete();
+
+			if (OS.getOS() != OS.WINDOWS) {
+				Set<PosixFilePermission> perms = new HashSet<>();
+				perms.add(PosixFilePermission.OWNER_READ);

```

Comment:
You should set permissions based on the settings from the tar file as I showed on the page I linked, do not hardcode permissions, also your permissions as you set them will not help anyway as you need to handle execute bits too.